Tests/CommandLineTests/ConvertActionTests.swift

```diff
@@ -3186,6 +3186,160 @@ class ConvertActionTests: XCTestCase {
             "/images/unit-test/image-name~dark@2x.png"
         ])
     }
+
+    func testExternalLinksInContentDontAffectNavigatorIndex() async throws {
+        let temporaryFolder = try createTemporaryDirectory()
+
+        let resolver: OutOfProcessReferenceResolver
+        do {
+            let executableLocation = temporaryFolder.appendingPathComponent("link-resolver-executable")
+
+            let externalBundleID = "com.external.test"
+            let externalSummary = LinkDestinationSummary(
+                kind: .class,
+                language: .swift,
+                relativePresentationURL: URL(string: "/documentation/TestBundle/SampleClass")!,
+                referenceURL: URL(string: "doc://\(externalBundleID)/documentation/TestBundle/SampleClass")!,
+                title: "SampleClass",
+                abstract: [.text("External abstract")],
+                availableLanguages: [.swift],
+                references: [],
+                variants: []
+            )
+            let encodedResponse = try String(decoding: JSONEncoder().encode(OutOfProcessReferenceResolver.ResponseV2.resolved(externalSummary)), as: UTF8.self)
+
+            try """
+        #!/bin/bash
+        echo '{"bundleIdentifier":"\(externalBundleID)","capabilities":0}'        # Write this resolver's bundle identifier
+        read                                                                    # Wait for docc to send a symbol USR
+        echo '\(encodedResponse)'                                             # Respond with the resolved link summary
+        """.write(to: executableLocation, atomically: true, encoding: .utf8)
+
+            // `0o0700` is `-rwx------` (read, write, & execute only for owner)
+            try FileManager.default.setAttributes([.posixPermissions: 0o0700], ofItemAtPath: executableLocation.path)
+            XCTAssert(FileManager.default.isExecutableFile(atPath: executableLocation.path))
+
+            resolver = try OutOfProcessReferenceResolver(processLocation: executableLocation, errorOutputHandler: { _ in })
```

Comment:
AFAIK all other tests that use an out-of-process resolver with an on-disk script like this wrap their code in `#if os(macOS)`. Those tests are specifically verifying the behavior of the `OutOfProcessReferenceResolver` but for this I think you just need to verify the behavior of an external source of documentation. For that you can make a simpler test if you use `TestExternalReferenceResolver` or `TestMultiResultExternalReferenceResolver`.